segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkManager.java

```diff
@@ -0,0 +1,1308 @@
+                lastChunk.setLength(chunkInfo.getLength());
+                segmentMetadata.setLength(segmentMetadata.getLastChunkStartOffset() + lastChunk.getLength());
+                txn.update(lastChunk);
+                log.debug("{} claimOwnership - Length of last chunk adjusted - segment={}, last chunk={}, Length={}.",
+                        logPrefix,
+                        segmentMetadata.getName(),
+                        lastChunk.getName(),
+                        chunkInfo.getLength());
+            }
+        }
+        // Update and commit
+        // If This instance is fenced this update will fail.
+        txn.update(segmentMetadata);
+        txn.commit();
+    }
+
+    @Override
+    public CompletableFuture<SegmentHandle> create(String streamSegmentName, SegmentRollingPolicy rollingPolicy, Duration timeout) {
+        checkInitialized();
+        return execute(() -> {
+            long traceId = LoggerHelpers.traceEnter(log, "create", streamSegmentName, rollingPolicy);
+            Timer timer = new Timer();
+
+            try (MetadataTransaction txn = metadataStore.beginTransaction()) {
+                // Retrieve metadata and make sure it does not exist.
+                SegmentMetadata oldSegmentMetadata = (SegmentMetadata) txn.get(streamSegmentName);
+                if (null != oldSegmentMetadata) {
+                    throw new StreamSegmentExistsException(streamSegmentName);
+                }
+
+                // Create a new record.
+                SegmentMetadata newSegmentMetatadata = SegmentMetadata.builder()
+                        .name(streamSegmentName)
+                        .maxRollinglength(rollingPolicy.getMaxLength() == 0 ? SegmentRollingPolicy.NO_ROLLING.getMaxLength() : rollingPolicy.getMaxLength())
+                        .ownerEpoch(this.epoch)
+                        .build();
+
+                newSegmentMetatadata.setActive(true);
+                txn.create(newSegmentMetatadata);
+                // commit.
+                txn.commit();
+
+                val retValue = SegmentStorageHandle.writeHandle(streamSegmentName);
+                Duration elapsed = timer.getElapsed();
+                log.debug("{} create - segment={}, rollingPolicy={}, latency={}.", logPrefix, streamSegmentName, rollingPolicy, elapsed.toMillis());
+                LoggerHelpers.traceLeave(log, "create", traceId, retValue);
+                return retValue;
+            } catch (StorageMetadataAlreadyExistsException ex) {
+                throw new StreamSegmentExistsException(streamSegmentName, ex);
+            } catch (StorageMetadataWritesFencedOutException ex) {
+                throw new StorageNotPrimaryException(streamSegmentName, ex);
+            }
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> write(SegmentHandle handle, long offset, InputStream data, int length, Duration timeout) {
```

Comment:
For "happy path" of appending to existing chunk this method is fast. It directly writes to the existing chunk. The metadata is not written to underlying store in this case. In system tests most writes are complete under 3 ms. (haven't run a perf run but don't expect a vastly different result. But will do it shortly.)

In case of need to add new chunk there is small extra overhead of creating a new chunk, updating and storing metadata.
For storage system segments, it is necessary to journal this change in system journal.